gnocchi/incoming/ceph.py

```diff
@@ -214,3 +217,32 @@ def process_measure_for_metrics(self, metric_ids):
                 self.ioctx.remove_omap_keys(op, tuple(keys.keys()))
                 self.ioctx.operate_write_op(op, str(sack),
                                             flags=self.OMAP_WRITE_FLAGS)
+
+    @contextlib.contextmanager
+    def process_measures_for_sack(self, sack):
+        measures = defaultdict(self._make_measures_array)
+        processed_keys = set()
+        omaps = self._list_keys_to_process(
+            sack, prefix=self.MEASURE_PREFIX + "_")
+        for k, v in six.iteritems(omaps):
+            try:
+                metric_id = uuid.UUID(k.split("_")[1])
+            except (ValueError, IndexError):
+                LOG.warning("Unable to parse measure object name %s",
```

Comment:
is this suppose to handle scenario where user put non-sense into measures path?

Reply:
Yes, just in case. It should never be hit.